pkg/registries/docker/config.go

```diff
@@ -51,11 +53,23 @@ func (c *Config) formatURL() string {
 
 // DefaultTransport returns the default transport based on the configuration.
 func DefaultTransport(cfg *Config) registry.Transport {
-	transport := proxy.RoundTripper()
+	defaultDialer := &net.Dialer{
+		Timeout:   registryDialerTimeout,
+		KeepAlive: 30 * time.Second,
+		DualStack: true,
+	}
```

Comment:
Can we merge this with `proxy/defaults.go` to have one exposed function like `DialerWithTimeout()`?

This way you can also remove your `proxy.WithDialContext` and change it to `proxy.WithDialTimeout` and not have each client have to create a dialer just for setting a timeout value.